packages/FormElement/src/FormElement.js

```diff
@@ -66,117 +70,91 @@ function FormElement(props) {
     isDisabled,
     isInline,
     isLabelVisuallyHidden,
-    isReadOnly,
     label,
     size,
     ...moreProps
   } = props;
 
   const extractedChildren = extractChildren(children, [
-    "FormElement.Description",
-    "FormElement.Error",
-    "FormElement.Help",
-    "FormElement.Instructions",
+    subComponentDisplayNames.Description,
+    subComponentDisplayNames.Error,
+    subComponentDisplayNames.Help,
+    subComponentDisplayNames.Instructions,
+    subComponentDisplayNames.Content,
   ]);
   const ariaDescriptionId = React.useRef(uuidv4()).current;
   const ariaErrorId = React.useRef(uuidv4()).current;
   const ariaInstructionsId = React.useRef(uuidv4()).current;
-  const hasError = !!extractedChildren["FormElement.Error"] && !!extractedChildren["FormElement.Error"].props.children;
   const uniqueInputId = React.useRef(uuidv4()).current;
-  const inputId = isNil(id) || id === "" ? uniqueInputId : id;
+  const hasError =
+    !!extractedChildren[subComponentDisplayNames.Error] &&
+    !!extractedChildren[subComponentDisplayNames.Error].props.children;
+  const idForLabel = isNil(id) || id === "" ? uniqueInputId : id;
   const refLabel = React.useRef(null);
 
+  const getClonedElement = (displayName, extraProps = {}) => {
+    return React.cloneElement(extractedChildren[displayName], extraProps);
+  };
+
   function renderFooter() {
     if (hasError) {
       return (
         <div role="alert">
-          {React.cloneElement(extractedChildren["FormElement.Error"], {
+          {getClonedElement(subComponentDisplayNames.Error, {
             id: ariaErrorId,
           })}
         </div>
       );
     }
 
-    if (extractedChildren["FormElement.Description"]) {
-      return React.cloneElement(extractedChildren["FormElement.Description"], {
+    if (extractedChildren[subComponentDisplayNames.Description]) {
+      return getClonedElement(subComponentDisplayNames.Description, {
         id: ariaDescriptionId,
       });
     }
 
     return null;
   }
 
-  function renderInstructions() {
-    if (extractedChildren["FormElement.Instructions"]) {
-      return React.cloneElement(extractedChildren["FormElement.Instructions"], {
-        id: ariaInstructionsId,
-      });
-    }
-    return null;
+  function renderContent() {
+    return getClonedElement(subComponentDisplayNames.Content, {
+      idForLabel,
+      refLabel,
+      ariaDescribedBy: `${ariaErrorId} ${ariaInstructionsId} ${ariaDescriptionId}`,
+    });
```

Comment:
This is the new `Content` subcomponent which will return a function to the consumer allowing them to use the props that can associate them with the label component

Reply:
Maybe we should add `required` too if `hasRequiredLabel` is `true`?

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/required

Reply:
This can be set by the consumer, I don't see the need to have this component figure out what the type of component the child is in order to know _if_ it should set `required` on it